front/components/assistant/markdown/CodeBlockWithExtendedSupport.tsx

```diff
@@ -0,0 +1,105 @@
+import { IconButton, SparklesIcon, WrenchIcon } from "@dust-tt/sparkle";
+import mermaid from "mermaid";
+import React, { useContext, useEffect, useRef, useState } from "react";
+
+import { CodeBlock } from "@app/components/assistant/markdown/CodeBlock";
+import { MarkdownContentContext } from "@app/components/assistant/markdown/MarkdownContentContext";
+import { classNames } from "@app/lib/utils";
+
+const MermaidGraph: React.FC<{ chart: string }> = ({ chart }) => {
+  const graphRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (graphRef.current) {
+      mermaid.initialize({ startOnLoad: false });
+      graphRef.current.innerHTML = chart;
+      void mermaid.init(undefined, graphRef.current);
+    }
+  }, [chart]);
+
+  return <div ref={graphRef} className="mermaid"></div>;
+};
+
+export function CodeBlockWithExtendedSupport({
+  children,
+  className,
+  inline,
+}: {
+  children?: React.ReactNode;
+  className?: string;
+  inline?: boolean;
+}) {
+  //   const { isValidMermaid, showMermaid } = useMermaidDisplay();
```

Comment:
?? No more Mermaid support?

Reply:
Yes, it's just moved below. I'll get rid of the comment.